src/map/skill.cpp

```diff
@@ -22672,48 +22658,229 @@ static bool skill_parse_row_nocastdb(char* split[], int columns, int current)
+
+			continue;
+		}
+
+		std::string item_name;
+
+		if (!this->asString(it, "Item", item_name))
+			return 0;
+
+		struct item_data *item = itemdb_search_aegisname(item_name.c_str());
+
+		if (item == nullptr) {
+			this->invalidWarning(it["Item"], "Item %s does not exist.\n", item_name.c_str());
+			return 0;
+		}
+
+		uint16 amount;
+
+		if (!isConsumed)
+			amount = 0;
+		else {
+			if (!this->asUInt16Rate(it, "Amount", amount, MAX_AMOUNT))
+				return 0;
+		}
+
+		entry->materials[item->nameid] = amount;
 	}
+	return 1;
+}
 
-	if (!itemdb_exists(nameid)) {
-		ShowError("skill_parse_row_producedb: Invalid item %u.\n", nameid);
-		return false;
+const std::string SkillProduceDatabase::getDefaultLocation() {
+	return std::string(db_path) + "/produce_db.yml";
+}
+
+/**
+ * Reads and parses an entry from the produce_db.
+ * @param node: YAML node containing the entry.
+ * @return count of successfully parsed rows
+ */
+uint64 SkillProduceDatabase::parseBodyNode(const YAML::Node &node) {
+	uint16 itemlv;
+
+	if (!this->asUInt16(node, "ItemLV", itemlv))
+		return 0;
+
+	std::shared_ptr<s_skill_produce_db> produce = this->find(itemlv);
+	bool exists = produce != nullptr;
+
+	if (!exists) {
+		if (!this->nodesExist(node, { "Recipe" }))
+			return 0;
+
+		produce = std::make_shared<s_skill_produce_db>();
+		produce->itemlv = static_cast<uint8>(itemlv);
 	}
 
-	skill_produce_db[id].nameid = nameid;
-	skill_produce_db[id].itemlv = atoi(split[2]);
-	skill_produce_db[id].req_skill = atoi(split[3]);
-	skill_produce_db[id].req_skill_lv = atoi(split[4]);
+	t_itemid nameid;
+	const YAML::Node &subNode = node["Recipe"];
+
+	for (const auto &subit : subNode) {
+		std::string produced_name;
+
+		if (this->nodeExists(subit, "Clear")) {
+			if (!this->asString(subit, "Clear", produced_name))
+				return 0;
+
+			struct item_data *item = itemdb_search_aegisname(produced_name.c_str());
+
+			if (item == nullptr) {
+				this->invalidWarning(subit["Clear"], "Item %s does not exist.\n", produced_name.c_str());
+				return 0;
+			}
+
+			nameid = item->nameid;
+
+			if (produce->data.erase(nameid) == 0)
+				this->invalidWarning(subit["Clear"], "Item %s was not defined.\n", produced_name.c_str());
+
+			continue;
+		}
+
+		if (!this->asString(subit, "Produced", produced_name))
+			return 0;
+
+		struct item_data *item = itemdb_search_aegisname(produced_name.c_str());
+
+		if (item == nullptr) {
+			this->invalidWarning(subit["Produced"], "Item %s does not exist.\n", produced_name.c_str());
+			return 0;
+		}
+
+		nameid = item->nameid;
+
+		bool id_exists = produce->data.count(nameid) != 0;
+		std::shared_ptr<s_skill_produce_db_entry> entry;
+
+		if (id_exists)
+			entry = produce->data[nameid];
+		else {
+			if (this->total_id >= MAX_SKILL_PRODUCE_DB) {
+				this->invalidWarning(subit["Produced"], "Maximum db entries reached. Increase MAX_SKILL_PRODUCE_DB.\n");
+				return 0;
+			}
```

Comment:
```suggestion
```
The MAX_SKILL_PRODUCE_DB is no longer needed since we are using dynamic storage.

Reply:
You're right, but I kept MAX_SKILL_PRODUCE_DB for now because it requires to update `clif.cpp` 
https://github.com/rathena/rathena/pull/6066/files?file-filters%5B%5D=.cpp#diff-e095af7d5069bf7fbe9926283903389918f6afe47ea6bae5a4880503b59c7d2cR6028